QUARKUS-5613.md

```diff
@@ -0,0 +1,85 @@
+## Links to the resources
+- JIRA issue: https://issues.redhat.com/browse/QUARKUS-5613
+- GH PRs related to the Jira issue: https://github.com/quarkusio/quarkus/pull/43283 https://github.com/quarkusio/quarkus/issues/42117 
+- Guides modified and related:
+  https://quarkus.io/guides/security-openid-connect-client-reference
+  https://quarkus.io/guides/security-oidc-code-flow-authentication
+  https://quarkus.io/guides/security-openid-connect-client-registration
+- TP/TD PRs:
+
+## Scope of the testing
+
+Our main testing scope is to ensure that the new response filter mechanism works as intended for various OIDC endpoints, ensuring that is possible intercept and handle responses
+from various OIDC endpoints.
+
+This complements the already available OidcRequestFilter functionality.
+
+Integration tests will be on the next module in our Test suite:    
+ security/keycloak-oidc-client-reactive-extended (https://github.com/quarkus-qe/quarkus-test-suite/tree/main/security/keycloak-oidc-client-reactive-extended).
+And the scope will be covered on areas related to:
+   - Security
+   - Logging /Tracing - Enabling TRACE level logs in properties will capture relevant info for our tests
+
+The test coverage will be verified for JVM, native mode, and Openshift.
+
+The testing include both OIDC Client (quarkus-oidc) and OIDC Client (quarkus-oidc-client) extensions.
+
+### What to Test - Scenarios
+
+#### Token Endpoint Response Filtering
+ - Verify that a custom OIDC response filter (TokenEndpointResponseFilter) is triggered immediately after a successful token refresh from Keycloak.
+ - Validation
+    * Ensure that filter logs messages contains "token have been refreshed".
+    * Verify 'OidcResponseContext' exposes the correct property grant_type=refresh_token
+    * Assert that the response includes "refresh_token"
```

Comment:
this `Validation` section proposes to verify what Keycloak sends, however that is Keycloak problem, not ours. We need to verify that validation of OIDC provider response is possible for body, headers, that request context fits the request filter and that logging is possible (as that is only documented thing...). So you can still test these scenarios to achieve that, but please rephrase whole `Validation` point. Thanks